src/Aspire.Hosting.Azure.Sql/AzureSqlExtensions.cs

```diff
@@ -210,21 +212,24 @@ private static void CreateSqlServer(
         },
         (infrastructure) =>
         {
-            // Creating a new SqlServer instance requires an administrator,
-            // so we need to create one here using the empty PrincipalId/PrincipalName
-            var principalIdParameter = new ProvisioningParameter(AzureBicepResource.KnownParameters.PrincipalId, typeof(string));
-            infrastructure.Add(principalIdParameter);
-            var principalNameParameter = new ProvisioningParameter(AzureBicepResource.KnownParameters.PrincipalName, typeof(string));
-            infrastructure.Add(principalNameParameter);
+            // Creating a new SqlServer instance requires an administrator. We create a dedicated user assigned managed identity.
+            var adminManagedIdentity = new UserAssignedIdentity("sqlServerAdminManagedIdentity")
```

Comment:
https://learn.microsoft.com/en-us/sql/relational-databases/security/authentication-access/azure-ad-authentication-sql-server-overview?view=sql-server-ver16#azure-active-directory-managed-identity

Feels like a deal breaker. 

I think we need to make sure that it's possible for the user to be able to manage the deployment database as well. Maybe that needs to be opt in?

Reply:
I guess we have this problem today.